community/cloud-foundation/templates/backend_service/tests/integration/backend_service.bats

```diff
@@ -0,0 +1,109 @@
+    RAND=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 10)
+    echo ${RAND} > "${RANDOM_FILE}"
+fi
+
+# Set variables based on the random string saved in the file.
+# envsubst requires all variables used in the example/config to be exported.
+if [[ -e "${RANDOM_FILE}" ]]; then
+    export RAND=$(cat "${RANDOM_FILE}")
+    DEPLOYMENT_NAME="${CLOUD_FOUNDATION_PROJECT_ID}-${TEST_NAME}-${RAND}"
+    # Replace underscores with dashes in the deployment name.
+    DEPLOYMENT_NAME=${DEPLOYMENT_NAME//_/-}
+    CONFIG=".${DEPLOYMENT_NAME}.yaml"
+    # test specific variables
+    export REGION="us-east1"
+    export RES_DESCRIPTION="This is the description"
+    export BC_DESCRIPTION="Backend description"
+    export TIMEOUT="35"
+    export ENABLE_CDN="false"
+    export SESSION="CLIENT_IP"
+    export REGIONAL_BALANCING_MODE="CONNECTION"
+    export GLOBAL_BALANCING_MODE="RATE"
+    export ZONE="us-central1-c"
+    export REGIONAL_BALANCING_SCHEME="INTERNAL"
+    export REGIONAL_BALANCING_PROTOCOL="TCP"
+    export GLOBAL_BALANCING_SCHEME="EXTERNAL"
+    export GLOBAL_BALANCING_PROTOCOL="HTTP"
+    export MAX_RATE="10000"
+fi
+
+########## HELPER FUNCTIONS ##########
+
+function create_config() {
+    echo "Creating ${CONFIG}"
+    envsubst < "templates/backend_service/tests/integration/${TEST_NAME}.yaml" > "${CONFIG}"
+}
+
+function delete_config() {
+    echo "Deleting ${CONFIG}"
+    rm -f "${CONFIG}"
+}
+
+function setup() {
+    # Global setup; executed once per test file.
+    if [ ${BATS_TEST_NUMBER} -eq 1 ]; then
+        create_config
+    fi
+
+    # Per-test setup steps.
+}
+
+function teardown() {
+    # Global teardown; executed once per test file.
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        delete_config
+    fi
+
+    # Per-test teardown steps.
+}
+
+
+@test "Creating deployment ${DEPLOYMENT_NAME} from ${CONFIG}" {
+    run gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" \
+        --config ${CONFIG} \
+        --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    [[ "$status" -eq 0 ]]
+}
+
+@test "Verifying global external backend service" {
+    run gcloud compute backend-services describe \
+        "global-external-backend-service-${RAND}" \
+        --project "${CLOUD_FOUNDATION_PROJECT_ID}" \
+        --global
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "balancingMode: ${GLOBAL_BALANCING_MODE}" ]]
+    [[ "$output" =~ "maxRate: ${MAX_RATE}" ]]
+    [[ "$output" =~ "loadBalancingScheme: ${GLOBAL_BALANCING_SCHEME}" ]]
+    [[ "$output" =~ "protocol: ${GLOBAL_BALANCING_PROTOCOL}" ]]
+}
+
+@test "Verifying regional internal backend service" {
+    run gcloud compute backend-services describe \
+        "regional-internal-backend-service-${RAND}" \
+        --project "${CLOUD_FOUNDATION_PROJECT_ID}" \
+        --region "${REGION}"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "balancingMode: ${REGIONAL_BALANCING_MODE}" ]]
+    [[ "$output" =~ "enableCDN: ${ENABLE_CDN}" ]]
```

Comment:
The test fails on "enableCDN", no such property comes up at the output.

Reply:
"When the load balancing scheme is INTERNAL, this field is not used."
https://cloud.google.com/compute/docs/reference/rest/v1/backendServices

Reply:
Interesting... The same test just succeeded for me. I'll troubleshoot

Reply:
I called the gcloud compute backend-services describe  "regional-internal-backend-service-${RAND}" manually and no anableCDN value showed up so it make sense for me why it fails. Do you get such a value?

Reply:
That's really cool: if I reorder assertions and move `enableCDN` down a bit, it also starts to fail locally. The output it tests against is always the same and it never has `enableCDN`. Beautiful.

Reply:
Fixed